kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,100 @@
+import os
+import logging
+import xmlschema
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+XML_SCHEMA = os.path.join('.', 'xml_schema')
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml, branch_name):
+    """validates XML file with existing schemas
+    :parsed_xml: parsed data from an XML file
+    """
+
+    failed, metadatacount, validcount = _validation_checks(report, parsed_xml, branch_name)
+
+    root_schema = os.path.join(XML_SCHEMA, 'addon.xsd')
+    if not _validate(parsed_xml, root_schema):
+        report.add(Record(PROBLEM, "Schema validation failed for root addon element"))
+        validcount -= 1
+
+    if metadatacount == 0:
+        report.add(Record(PROBLEM, 'Metadata missing/occurred more than once'))
+
+    if validcount == 1:
+        report.add(Record(INFORMATION, "Valid XML file found"))
+    elif failed:
+        report.add(Record(PROBLEM, "Schema validation failed for the following points: %s " % ''.join(failed)))
+
+
+def _validation_checks(report: Report, parsed_xml, branch_name):
+    """Perform certain checks on the XML elements and extension point
+    """
+    metadatacount = 0
+    validcount = 1
+    failed = []
+    valid_points = {'xbmc.python.module': 'script.xsd', 'kodi.addon.metadata': 'metadata.xsd',
```

Comment:
please keep one `key:value` per line to improve readability. Some keys are repeated (kodi.addon.metadata and xbmc.python.module)

Reply:
also sort them somehow, best would be sort by value and then per key